### ISSUE TITLE
Error: Cannot find module 'repromise'

### ISSUE THREAD

**Issue by pdehaan** [tracker1/repromise#1]
I'm getting the following error when trying to use repromise:

> _Error: Cannot find module 'repromise'_

---
### ./package.json

``` js
$ cat package.json
{
  "name": "repromise-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Peter deHaan <peter@deseloper.com> (http://nodeexamples.com/)",
  "license": "WTFPL",
  "dependencies": {
    "bluebird": "2.9.24",
    "repromise": "3.0.3"
  },
  "peerDependencies": {
    "i-promise": "1.1.0"
  }
}
```
### % npm ls

``` sh
$ npm ls --depth 1
repromise-test@1.0.0 /Users/pdehaan/dev/tmp/repromise-test
├── bluebird@2.9.24
├── i-promise@1.1.0
└─┬ repromise@3.0.3
  └── debug@2.1.3
```
### ./index.js

``` js
$ cat index.js
var P = require('bluebird');
require('i-promise/config').use(P);

var repromise = require('repromise');

repromise(function () {
  return new Promise(function (resolve, reject) {
    resolve(1);
  });
}).then(function (value) {
  console.log(value);
}).catch(function (err) {
  console.error(err);
});
```
### output:

``` sh
$ node index

module.js:340
    throw err;
          ^
Error: Cannot find module 'repromise'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/pdehaan/dev/tmp/repromise-test/index.js:4:17)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:929:3
```

**Comment by pdehaan** [comment]
If I understand correctly, it doesn't look like ./dist/\* isn't being generated.

``` sh
$ tree node_modules/repromise/
node_modules/repromise/
├── LICENSE
├── README.md
├── node_modules
│   └── debug
│       ├── History.md
│       ├── Makefile
│       ├── Readme.md
│       ├── bower.json
│       ├── browser.js
│       ├── component.json
│       ├── debug.js
│       ├── node.js
│       ├── node_modules
│       │   └── ms
│       │       ├── LICENSE
│       │       ├── README.md
│       │       ├── index.js
│       │       └── package.json
│       └── package.json
├── package.json
└── src
    ├── get-retry-counts.js
    └── index.js

5 directories, 18 files
```

---

Note that [`"main": "dist/index.js"`](https://github.com/tracker1/repromise/blob/66f9ec7f0eb4f88b26b503c401aa33fae6610895/package.json#L5) is specified in the package.json file, but there is no /dist/ directory in my /node_modules/repromise/ directory.

``` js
$ cat node_modules/repromise/package.json
{
  "name": "repromise",
  "version": "3.0.3",
  "description": "JavaScript Promise Retry Module",
  "main": "dist/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/tracker1/repromise.git"
  },
  "keywords": [
    "promise",
    "retry",
    "es6",
    "promises"
  ],
  "author": {
    "name": "Michael J. Ryan",
    "email": "tracker1@gmail.com"
  },
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/tracker1/repromise/issues"
  },
  "homepage": "https://github.com/tracker1/repromise",
  "config": {
    "MOCHA_OPTS": "--recursive spec/",
    "ISPARTA_OPTS": "--report html",
    "COVERAGE_OPTS": "--statements 100 --functions 100 --branches 100 --lines 100"
  },
  "scripts": {
    "clean": "rimraf ./dist",
    "build": "babel ./src --stage 0 --source-maps-inline -d ./dist",
    "lint": "npm run lint-src && npm run lint-test",
    "lint-src": "eslint src",
    "lint-test": "eslint spec --global describe,it,beforeEach,afterEach --rule no-unused-expressions:false ",
    "unit": "babel-node -r node_modules/.bin/_mocha $npm_package_config_MOCHA_OPTS",
    "coverage": "babel-node -r node_modules/.bin/isparta cover $npm_package_config_ISPARTA_OPTS node_modules/.bin/_mocha -- $npm_package_config_MOCHA_OPTS",
    "coverage-check": "node node_modules/isparta/node_modules/istanbul/lib/cli check-coverage $npm_package_config_COVERAGE_OPTS",
    "show-report": "node scripts/open-coverage-report",
    "test": "npm run lint && npm run coverage && (npm run coverage-check || node scripts/open-coverage-report)",
    "publish-major": "npm run build && npm version major && npm publish && git push origin master && git push --tags",
    "publish-minor": "npm run build && npm version minor && npm publish && git push origin master && git push --tags",
    "publish-patch": "npm run build && npm version patch && npm publish && git push origin master && git push --tags"
  },
  "peerDependencies": {
    "i-promise": "^1.1.0"
  },
  "dependencies": {
    "debug": "^2.1.3"
  },
  "devDependencies": {
    "eslint": "^0.18.0",
    "babel": "^4.7.12",
    "chai": "^2.1.1",
    "isparta": "^2.2.0",
    "mocha": "^2.2.1",
    "open": "0.0.5",
    "proxyquire": "^1.4.0",
    "sinon": "^1.13.0",
    "sinon-chai": "^2.7.0"
  },
  "gitHead": "66f9ec7f0eb4f88b26b503c401aa33fae6610895",
  "_id": "repromise@3.0.3",
  "_shasum": "cc776245844072e0c6ef952a7d8274639f1acf69",
  "_from": "repromise@3.0.3",
  "_npmVersion": "2.7.4",
  "_nodeVersion": "1.6.3",
  "_npmUser": {
    "name": "tracker1",
    "email": "tracker1@gmail.com"
  },
  "maintainers": [
    {
      "name": "tracker1",
      "email": "tracker1@gmail.com"
    }
  ],
  "dist": {
    "shasum": "cc776245844072e0c6ef952a7d8274639f1acf69",
    "tarball": "http://registry.npmjs.org/repromise/-/repromise-3.0.3.tgz"
  },
  "directories": {},
  "_resolved": "https://registry.npmjs.org/repromise/-/repromise-3.0.3.tgz",
  "readme": "ERROR: No README data found!"
}
```

**Comment by tracker1** [comment]
Thanks, should be fixed in 3.1.0
Accidentally put the `dist/` directory in `.npmignore` not `.gitignore`

Rather embarrassing.
